api/datastore/postgres/postgres.go

```diff
@@ -411,15 +401,18 @@ func scanRoute(scanner rowScanner, route *models.Route) error {
 		&headerStr,
 		&configStr,
 	)
+	if err != nil {
+		return err
+	}
 
 	if headerStr == "" {
 		return models.ErrRoutesNotFound
 	}
```

Comment:
Can anyone explain this one?  Why must the `headerStr` field be populated?  Why is this seemingly bad-data reported the same as a route not being found?  Note that it is not only called from `GetRoute` but `GetRoutes` as well, which does some questionable error discarding.